src/main/java/com/jfrog/ide/idea/ui/LocalComponentsTree.java

```diff
@@ -27,20 +31,39 @@ public static LocalComponentsTree getInstance(@NotNull Project project) {
         return project.getService(LocalComponentsTree.class);
     }
 
+
     private void appendProjectWhenReady(FileTreeNode filteredRoot) {
         ApplicationManager.getApplication().invokeLater(() -> appendProject(filteredRoot));
     }
 
     @Override
     public void reset() {
-        fileNodes = new ArrayList<>();
+        fileNodes = new HashMap<>();
         super.reset();
     }
 
+    @Override
+    public void applyFiltersForAllProjects() {
+        setModel(null);
+        // TODO: I added the reset here, but it might not be the right solution, and i'm not even sure if it's needed. If it's removed or moved from here, pay attention to the commented code in addOrReplace below.
+        reset();
+        for (FileTreeNode node : fileNodes.values()) {
+            appendProjectWhenReady(node);
+        }
+    }
+
     public void addScanResults(List<FileTreeNode> fileTreeNodes) {
         for (FileTreeNode node : fileTreeNodes) {
-            fileNodes.add(node);
-            appendProjectWhenReady(node);
+            var fileNode = fileNodes.get(node.getPath().toString());
+            if (fileNode != null) {
+                List newChildren = Collections.list(fileNode.children());
+                for (var child : newChildren) {
+                    fileNode.add((MutableTreeNode) child);
+                }
```

Comment:
More efficient to not create a copy list:
```suggestion
                Enumeration<TreeNode> children = fileNode.children();
                while (children.hasMoreElements()) {
                    fileNode.add((MutableTreeNode) children.nextElement());
                }
```